lib/io/rdf/sparql/sparql-query-rdf-source.ts

```diff
@@ -0,0 +1,12 @@
+import {HttpFetch} from "../../fetch/fetch-api";
+import {RdfMemorySource} from "../rdf-source-base";
+import {fetchRdfQuads} from "../http/http-adapter";
+
+export class SparqlQueryRdfSource extends RdfMemorySource {
+
+  async query(httpFetch: HttpFetch, endpoint: string, query: string): Promise<void> {
+    const url = endpoint + "?format=" + encodeURIComponent("text/turtle") + "&query=" + encodeURIComponent(query);
+    this.quads = await fetchRdfQuads(httpFetch, url);
```

Comment:
From the user perspective, you create a repository and you execute the query, then you start reading the results. But if during processing the the results you query again, you loose the original data. I would suggests either to add the statements, or specify the query in ctor - making the source fixed for a single httpFetch, endpoint and query.